capabilities/src/execute_task_solution_capability.cpp

```diff
@@ -142,52 +142,75 @@ bool ExecuteTaskSolutionCapability::constructMotionPlan(const moveit_task_constr
 		state = scene->getCurrentState();
 	}
 
-	plan.plan_components.reserve(solution.sub_trajectory.size());
-	for (size_t i = 0; i < solution.sub_trajectory.size(); ++i) {
+	plan.plan_components_.reserve(solution.sub_trajectory.size());
+	auto make_executable_trajectory =
+	    [&](const moveit_task_constructor_msgs::msg::SubTrajectory& sub_traj, const std::string& description,
+	        const boost::function<bool(const plan_execution::ExecutableMotionPlan*)>& on_success_callback) {
+		    plan.plan_components_.emplace_back();
+		    plan_execution::ExecutableTrajectory& exec_traj = plan.plan_components_.back();
+		    exec_traj.description_ = description;
+		    const moveit::core::JointModelGroup* group = nullptr;
+		    {
+			    std::vector<std::string> joint_names(sub_traj.trajectory.joint_trajectory.joint_names);
+			    joint_names.insert(joint_names.end(), sub_traj.trajectory.multi_dof_joint_trajectory.joint_names.begin(),
+			                       sub_traj.trajectory.multi_dof_joint_trajectory.joint_names.end());
+			    if (!joint_names.empty()) {
+				    group = findJointModelGroup(*model, joint_names);
+				    if (!group) {
+					    RCLCPP_ERROR_STREAM(LOGGER, "Could not find JointModelGroup that actuates {"
+					                                    << boost::algorithm::join(joint_names, ", ") << "}");
+					    return false;
+				    }
+				    RCLCPP_DEBUG(LOGGER, "Using JointModelGroup '%s' for execution", group->getName().c_str());
+			    }
+		    }
+		    exec_traj.trajectory_ = std::make_shared<robot_trajectory::RobotTrajectory>(model, group);
+		    exec_traj.trajectory_->setRobotTrajectoryMsg(state, sub_traj.trajectory);
+		    exec_traj.controller_names_ = sub_traj.execution_info.controller_names;
+
+		    /* todo add action feedback and markers */
+		    exec_traj.effect_on_success_ = on_success_callback;
+		    return true;
+	    };
+
+	auto make_apply_planning_scene_diff_cb = [this](const std::vector<moveit_msgs::msg::PlanningScene>& scene_diffs) {
+		return
+		    [this, scene_diffs = std::move(scene_diffs)](const plan_execution::ExecutableMotionPlan* /*plan*/) mutable {
+			    for (auto& scene_diff : scene_diffs) {
+				    if (!moveit::core::isEmpty(scene_diff)) {
+					    /* RCLCPP_DEBUG_STREAM(LOGGER, "apply effect of " << description); */
+					    scene_diff.robot_state.joint_state = sensor_msgs::msg::JointState();
+					    scene_diff.robot_state.multi_dof_joint_state = sensor_msgs::msg::MultiDOFJointState();
+					    if (!context_->planning_scene_monitor_->newPlanningSceneMessage(scene_diff))
```

Comment:
Comment of @v4hn:

> This is a fixup for https://github.com/ros-planning/moveit_task_constructor/commit/72afdd22d2a941350294c33fae3cccaf585f5b44 , please merge the commits.
Also, you should set robot_state.is_diff = true explicitly if you empty the robot_state fields and do it before you check for isEmpty because the set robot state will often be the only reason why the scene diff is not empty.